### DISCUSSION TITLE
Getting real-time CF position data using cflib/Crazyswarm ROS

### DISCUSSION THREAD

**Discussion by coenwerem** [USC-ACTLab/crazyswarm#659] [Q&A] [answered]
Hello all,

I recently started working on a swarm project with the Crazyflie 2.1 quadcopter. Using the excellent and very helpful Crazyswarm ROS package, I have been able to successfully fly 4 CFs to custom waypoints by modifying several scripts from the Crazyswarm ROS package. However, for the script to work as intended, I currently have to specify the real starting positions of the CFs in the inertial frame using the allCrazyflies.yaml file. What I want to be able to do is to fly the 4 CFs to custom points in 3D space without hardcoding their initial positions. That is, can I programmatically obtain the real-time position data of the CFs using either the Crazyswarm ROS package or cflib to use that in the script instead? I have read through the Crazyswarm and cflib docs without any luck. I also tried using the methods in the `position_hl_commander.py` cflib script, but don't seem to be getting anywhere as I am running into some errors. 

I am using the V1 Lighthouse Positioning System for external localization.

Any help will be appreciated!

Many thanks

**Comment by whoenig** [comment, marked as answer]
LightHouse is decentralized tracking, so the PC does not know by default the current positions. You can set `enable_logging` and `enable_logging_pose` to true in the launch file, which will create topics like `cf<id>/pose` containing the current state information. In your script, you can then use `position()` or subscribe to the topic manually if you need velocities as well.

**Reply by coenwerem** [reply]
Thank you so much for the prompt response! I have those two parameters set to True in the `hover_swarm` launch file. I did not know about the cf<id>/pose topic. I will try that out. 

Thank you.
